File: 백준/Gold/11657. 타임머신/타임머신.ts
function bellmanFord() {
  const INF = 5000001;
  const dist: number[] = Array(N+1).fill(INF);
  dist[1] = 0;
  
  for (let u=1; u<N+1; u++) {
    for (const [a, b, c] of graph) {
      if (dist[a] === INF) {continue}; // unreachable
      if (dist[b] > dist[a] + c) {
        dist[b] = dist[a] + c;
      }
    }
  }

  for (let u=1; u<N+1; u++) {
    for (const [a, b, c] of graph) {
      if (dist[a] === INF) {continue};
      if (dist[b] > dist[a] + c) {
        return -1 // cycle
      }
    }
  }

  let ans = "";
  for (let i=2; i<N+1; i++){
    ans += `${dist[i] === INF ? -1 : dist[i]} `;
  }
  return ans
}


const input: string[] = require("fs").readFileSync(process.platform==="linux"?0:"input.txt").toString().trim().split("\n");
const [N, M] = input[0].split(/\s/).map(Number);
const graph: number[][] = [];
for (let i=1; i<M+1; i++) {
  graph.push(input[i].split(/\s/).map(Number));
}

console.log(bellmanFord());

File: 백준/Gold/11657. 타임머신/타임머신.py
def bellmanFord():
  INF = 5000001
  dist = [INF] * (N+1)
  dist[1] = 0
  
  for _ in range(1, N):
    for a, b, c in graph:
      if dist[a] != INF and dist[b] > dist[a] + c:
        dist[b] = dist[a] + c

  for _ in range(1, N):
    for a, b, c in graph:
      if dist[a] != INF and dist[b] > dist[a] + c:
        return -1

  ans = ""
  for i in range(2, N+1):
    ans += "-1\n" if dist[i] == INF else f"{dist[i]}\n"
  return ans


N, M = map(int, input().split())
graph = [(a, b, c) for _ in range(M) for a, b, c in [map(int, input().split())]]
print(bellmanFord())